### PR TITLE
caf: ble_adv: Improve initialization chain

### PR DESCRIPTION
**Pull request by MarekPieta**
The module must wait for Bluetooth settings load and ble_bond module readiness before the advertising is started. If any of these dependencies is not enabled, the module must not wait for it.

Jira: NCSDK-17522

### CODE REVIEW

File: subsys/caf/modules/ble_adv.c [hunk 1]
```diff
@@ -410,18 +410,15 @@ static void init(void)
...
 static void start(void)
 {
 	int err = ble_adv_start(true);
 
-	if (err) {
+	if (!err) {
+		module_set_state(MODULE_STATE_READY);
+	} else {
 		module_set_state(MODULE_STATE_ERROR);
 	}
```
Comment (kapi-no) [inline subsys/caf/modules/ble_adv.c:423]: If it is possible, please limit the number of negations:

```suggestion
	if (err) {
		module_set_state(MODULE_STATE_ERROR);
	} else {
		module_set_state(MODULE_STATE_READY);
	}
```

**Review by kapi-no — COMMENTED** [review]
shouldn't we add entry to release notes?

File: subsys/caf/modules/Kconfig.ble_adv [hunk 2]
```diff
@@ -8,6 +8,7 @@ menuconfig CAF_BLE_ADV
 	bool "Enable Bluetooth LE advertising"
 	depends on BT_PERIPHERAL
 	depends on CAF_BLE_COMMON_EVENTS
+	depends on CAF_BLE_STATE
```
Comment (kapi-no) [inline subsys/caf/modules/Kconfig.ble_adv:11]: This line seems like a fix for the dependency description. If so, please move it to a separate commit

File: subsys/caf/modules/ble_adv.c [hunk 3]
```diff
@@ -473,7 +470,7 @@ static void disconnect_peer(struct bt_conn *conn)
...
 }
 
-static void ble_initialized(void)
+static void ble_ready(void)
```
Comment (kapi-no) [inline subsys/caf/modules/ble_adv.c:473]: Please avoid renames in the same commit that are not related to the commit change.

(The ```ble_ready``` name seems better because it is called in response to the READY event. However, this part worked like this before the main PR change and should be corrected in a separate commit).

**Review by kapi-no — COMMENTED** [review]
A few comments regarding the commit separation

**Comment by MarekPieta** [comment]
> shouldn't we add entry to release notes?

I will open a separate, follow-up PR for doc changes.

File: subsys/caf/modules/ble_adv.c (hunk 1 above)
Comment (MarekPieta) [reply subsys/caf/modules/ble_adv.c:423]: I initially preferred to keep the happy-path at the top, but changed according to your suggestions

File: subsys/caf/modules/ble_adv.c (hunk 3 above)
Comment (MarekPieta) [reply subsys/caf/modules/ble_adv.c:473]: The rename is related to the change. When ble_bond was disabled we used to wait only for BLE initialization, now we properly wait for BLE readiness to start advertising (BLE becomes ready after it's initialized and settings are loaded).

File: subsys/caf/modules/ble_adv.c [hunk 4]
```diff
@@ -498,23 +495,51 @@ static void ble_initialized(void)
 	started = true;
 }
 
+static inline void get_req_modules(struct module_flags *mf)
+{
+	BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_SETTINGS) ||
+		     IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER),
+		     "Bluetooth settings enabled, but settings loader module is disabled");
```
Comment (alstrzebonski) [inline subsys/caf/modules/ble_adv.c:502]: For me `BUILD_ASSERT(!(IS_ENABLED(CONFIG_BT_SETTINGS) && !IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER)), ...);` would be much more clear.

File: subsys/caf/modules/ble_adv.c [hunk 5]
```diff
@@ -498,23 +495,51 @@ static void ble_initialized(void)
...
+
+	module_flags_set_bit(mf, MODULE_IDX(ble_state));
+#if CONFIG_CAF_SETTINGS_LOADER
+	module_flags_set_bit(mf, MODULE_IDX(settings_loader));
+#endif
+#if CONFIG_CAF_BLE_ADV_BLE_BOND_SUPPORTED
+	module_flags_set_bit(mf, MODULE_IDX(ble_bond));
+#endif
```
Comment (alstrzebonski) [inline subsys/caf/modules/ble_adv.c:510]: Shouldn't we use `if IS_ENABLED(...)` instead of `#if ...`?

**Review by alstrzebonski — COMMENTED** [review]
Initial review. Looks good apart from comments that I posted, but I will need to tak a closer look to understand what really happens in this module. It seems complicated.

File: subsys/caf/modules/ble_adv.c (hunk 5 above)
Comment (MarekPieta) [reply subsys/caf/modules/ble_adv.c]: Changed. I needed to double check to ensure that code is opt-out by compiler in that case to prevent linker errors.

File: subsys/caf/modules/ble_adv.c (hunk 4 above)
Comment (kapi-no) [reply subsys/caf/modules/ble_adv.c]: Or simply:

```BUILD_ASSERT(IS_ENABLED(CONFIG_BT_SETTINGS) == IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER), ...);```

**Review by kapi-no — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Comment by MarekPieta** [comment]
Rebased and force pushed to retrigger CI

**Review by rakons — APPROVED** [review]

**Review by alstrzebonski — APPROVED** [review]
